### PR TITLE
[#75238] Jira Migrator cannot import a user with non-alphanumeric characters in their name

### PR DESCRIPTION
**Pull request by as-op**
# Ticket

https://community.openproject.org/wp/75238

# What are you trying to achieve?
Jira import is stopped when a (split) Jira user name contains chars not allowed by the user model.
e.g. "[BOT] Ticketron"
Import should continue.

# What approach did you choose and why?

Filter out these characters and continue with importing
"[BOT] Ticketron" ⇒ first name: "BOT"; last name "Ticketron"

Edge case, name consists only of special characters
"[]" ⇒ first name: "Unknown"; last name "Unknown"

# Merge checklist

- [x] Added/updated tests
- [x] Tested major browsers (Chrome, Firefox, Edge, ...)

### CODE REVIEW

File: app/models/import/jira_user.rb [hunk 1]
```diff
@@ -56,8 +58,12 @@ def try_to_find_existing_op_users
 
     private
 
+    def sanitize_name(name)
+      name.chars.grep(User::VALID_NAME_REGEX).join.strip
```
Comment (ba1ash) [inline app/models/import/jira_user.rb]:
```suggestion
      name.gsub(User::INVALID_NAME_REGEX, '')
```
I have the following suggestion. It avoids some allocations(probably not significant) and it is simpler in my opinion. But we need to define a few constants in user.rb:
```ruby
  VALID_NAME_CHARS = "\\d\\p{Alpha}\\p{Mark}\\p{Space}\\p{Emoji}''´\\-_.,@()+&*–"
  INVALID_NAME_REGEX = /[^#{VALID_NAME_CHARS}]/
  VALID_NAME_REGEX   = /\A[#{VALID_NAME_CHARS}]+\z/
```
WDYT?

Comment (as-op) [reply app/models/import/jira_user.rb]: Like it a lot. Don't know why I did not want to change the regex, as there is no good reason for that.

I changed it.

**Review by ba1ash — APPROVED** [review]
